auth/tasks.py

```diff
@@ -50,51 +61,79 @@ def update_group_and_perms(department, created):
+        # 将当前department的所有叶子结点返回
+        childs = []
+        child_departments = Department.child_departments.all()
+        if child_departments:
+            for child_department in child_departments:
+                childs.extend(find_all_child_department(child_department))
+        else:
+            childs.extend([super_department])
+        return childs
+    try:
+        for raw_department in raw_departments:
+            department, created = Department.objects.get_or_create(
+                raw_department_id=raw_department.dwid,
+                defaults={'name': raw_department.dwmc})
+            update_group_and_perms(department, created)
+            updated = False
+            # 同步隶属单位
+            if (department.super_department is None or
+                    department.super_department.raw_department_id !=
+                    raw_department.lsdw):
+                super_department_name = DepartmentInformation.objects.get(
+                    dwid=raw_department.lsdw).dwmc
+                super_department, created = Department.objects.get_or_create(
+                    raw_department_id=raw_department.lsdw,
+                    defaults={'name': super_department_name}
+                )
+                update_group_and_perms(super_department, created)
+                # 将老师从原有group中删除
+                if department.super_department:
+                    child_departments = find_all_child_department(
+                        department.super_department)
+                    for department in child_departments:
+                        teachers = User.objects.filter(department=department)
+                        for teacher in teachers:
+                            update_user_groups(teacher.groups.remove,
+                                               department, dlut)
+                            teacher.department = None
+                            teacher.save()
+                department.super_department = super_department
+                updated = True
+            # 同步单位类型
+            if department.department_type != raw_department.dwlx:
+                department.department_type = raw_department.dwlx
+                updated = True
+
+            # 同步单位名称
+            if department.name != raw_department.dwmc:
+                department.name = raw_department.dwmc
+                updated = True
+            if updated:
+                department.save()
+            if dlut in (department.super_department,
+                        department.super_department.super_department):
+                # 校区和二级部门的administrative为本身
+                department_id_to_administrative[department.id] = department
+            else:
+                # 非二级部门的administrative暂为superdepartment
+                department_id_to_administrative[department.id] = (
+                    department.super_department
+                )
+    except Exception:
+        prod_logger.exception('部门信息更新失败,单位号:%s',
```

Comment:
The log shows no information about what exactly the error is, consider add more useful information for debug.
```python
try:
    ....
except Exception as exc:
    prod_logger.exception('...%s', ..., exc)
```